### PR TITLE
Domain dashboard card: Remove feature flag

### LINKED ISSUE #18725: Domain Dashboard Card: Remove related code
With the inclusion of Plans related card, Domain Dashboard Card is no longer needed:

- [ ] Remove dashboard card UI code
- [ ] Remove the feature flag
- [ ] Remove presentation criteria

### PR DESCRIPTION
**Pull request by ravishanker**
Fixes #18725 

Removing domain dashboard card, and clean up all related as it is superseded by Plans project.

To test:

- Launch JP app
- `Go` to Me -> Debug settings
- `Verify` no dashboard_card_domain feature flag is present
- `Go` to Home
- `Verify` **Home**, **Menu** options on dashboard all work as before

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ravishanker** [comment]
@AjeshRPai - FYI

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/DashboardTests.kt [hunk 1]
```diff
@@ -21,15 +21,6 @@ class DashboardTests : BaseTest() {
...
     }
 
-    @Test
-    fun e2eDomainsCardNavigation() {
-        MySitesPage()
-            .scrollToDomainsCard()
-            .assertDomainsCard()
-            .tapDomainsCard()
-            .assertDomainsScreenLoaded()
-    }
```
Comment (irfano) [inline WordPress/src/androidTest/java/org/wordpress/android/e2e/DashboardTests.kt]: After removing this, `DomainsScreen` became unused, we can remove it too.

File: WordPress/build.gradle [hunk 2]
```diff
@@ -146,7 +146,8 @@ android {
         buildConfigField "boolean", "JETPACK_INSTALL_FULL_PLUGIN", "false"
         buildConfigField "boolean", "ENABLE_BLAZE_FEATURE", "false"
         buildConfigField "boolean", "WP_INDIVIDUAL_PLUGIN_OVERLAY", "false"
-        buildConfigField "boolean", "DASHBOARD_CARD_DOMAIN", "false"
+        buildConfigField "boolean", "DASHBOARD_CARD_PAGES", "false"
+        buildConfigField "boolean", "DASHBOARD_CARD_ACTIVITY_LOG", "false"
```
Comment (irfano) [inline WordPress/build.gradle]: Why did we add `DASHBOARD_CARD_PAGES` and `DASHBOARD_CARD_ACTIVITY_LOG`?

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt [hunk 3]
```diff
@@ -381,14 +381,6 @@ sealed class MySiteCardAndItem(open val type: Type, open val activeQuickStartIte
                     }
                 }
 
-                data class DashboardDomainCard(
-                    val title: UiString?,
-                    val subtitle: UiString?,
-                    val onClick: ListItemInteraction,
-                    val onHideMenuItemClick: ListItemInteraction,
-                    val onMoreMenuClick: ListItemInteraction,
-                ) : DashboardCard(dashboardCardType = DashboardCardType.DASHBOARD_DOMAIN_CARD)
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt:379]: We can also remove `DashboardCardType.DASHBOARD_DOMAIN_CARD`, right?

**Review by irfano — CHANGES_REQUESTED** [review]
We can also remove
- `AppPrefsWrapper.getShouldHideDashboardDomainCard()` and `AppPrefsWrapper.setShouldHideDashboardDomainCard()`.
- "dashboard_card_domain" in feature-flags.json.
- `Partial State.CustomDomainsAvailable`

File: WordPress/build.gradle (hunk 2 above)
Comment (ravishanker) [reply WordPress/build.gradle]: Probably, came in through trunk merge.  Removed, thanks

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java [hunk 4]
```diff
@@ -1650,18 +1650,6 @@ public static void setShouldHidePromoteWithBlazeCard(long siteId, final boolean
...
-    public static Boolean getShouldHideDashboardDomainCard(long siteId) {
-        return prefs().getBoolean(getSiteIdHideDashboardDomainCardKey(siteId), false);
-    }
-
-    public static void setShouldHideDashboardDomainCard(long siteId, final boolean isHidden) {
-        prefs().edit().putBoolean(getSiteIdHideDashboardDomainCardKey(siteId), isHidden).apply();
-    }
-
-    @NonNull private static String getSiteIdHideDashboardDomainCardKey(long siteId) {
-        return DeletablePrefKey.SHOULD_HIDE_DASHBOARD_DOMAIN_CARD.name() + siteId;
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java:1662]: We can also remove `DeletablePrefKey. SHOULD_HIDE_DASHBOARD_DOMAIN_CARD`.

**Review by irfano — APPROVED** [review]
Thank you for addressing my comments. I've added one last suggestion and approved the PR. Then you can merge it. LGTM 👍🏻

**Pull request merged**